pandas/core/series.py

```diff
@@ -206,7 +206,11 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
                     data = data.astype(dtype)
                 else:
                     # need to copy to avoid aliasing issues
-                    data = data._values.copy()
```

Comment:
wouldn't passing deep=True simply solve this?

Reply:
I think the problem only occurs with tz-aware cases.

Reply:
this is not a good solution here, to special case things. pass ``deep=True`` here; if this doesn't work, then the copy constructor is broken.

Reply:
This bug is caused by the return value of _values. In datetimes.py, _values returns the DatetimeIndex itself for tz-aware and ndarray for tz-naive.
```
    @property
    def _values(self):
        # tz-naive -> ndarray
        # tz-aware -> DatetimeIndex
        if self.tz is not None:
            return self
        else:
            return self.values

```
That 's why  this problem only occurs with tz-aware cases as jbrockmendel said.
```

In [2]: dti1 = pd.date_range('2016-01-01', periods=10, tz='US/Pacific')

In [3]: dti2 = pd.date_range('2016-01-01', periods=10)

In [4]: ser1 = pd.Series(dti1)

In [5]: ser2 = pd.Series(dti2)

In [6]: ser1[::3]=pd.NaT

In [7]: ser2[::3]=pd.NaT

In [8]: dti1
Out[8]: 
DatetimeIndex([                      'NaT', '2016-01-02 00:00:00-08:00',
               '2016-01-03 00:00:00-08:00',                       'NaT',
               '2016-01-05 00:00:00-08:00', '2016-01-06 00:00:00-08:00',
                                     'NaT', '2016-01-08 00:00:00-08:00',
               '2016-01-09 00:00:00-08:00',                       'NaT'],
              dtype='datetime64[ns, US/Pacific]', freq='D')

In [9]: dti2
Out[9]: 
DatetimeIndex(['2016-01-01', '2016-01-02', '2016-01-03', '2016-01-04',
               '2016-01-05', '2016-01-06', '2016-01-07', '2016-01-08',
               '2016-01-09', '2016-01-10'],
              dtype='datetime64[ns]', freq='D')

```